client/lib/domains/dns/index.js

```diff
@@ -126,7 +130,57 @@ function getFieldWithDot( field ) {
 	return ( typeof field === 'string' && field.match( /^([a-z0-9-_]+\.)+\.?[a-z]+$/i ) ) ? field + '.' : field;
 }
 
-module.exports = {
-	validateAllFields,
-	getNormalizedData
+function isWpcomRecord( record ) {
+	return startsWith( 'wpcom:', record.id );
+}
+
+function filterRootARecords( domain, records ) {
+	return filter( records, {
+		type: 'A',
+		name: `${domain}.`
+	} );
+}
+
+function removeDuplicateWpcomRecords( domain, records ) {
+	const wpcomRecord = find( filterRootARecords( domain, records ), isWpcomRecord ),
+		customRecord = find( records, { type: 'A', name: '' } );
+
+	let result;
```

Comment:
The code I've proposed:
```
    if ( wpcomRecord && customRecord ) {
        return without( records, wpcomRecord );
    }

    return records;
```
does not violate that rule. It's actually the first example in the "The follow patterns are not considered problems:" section of [its documentation](http://eslint.org/docs/rules/no-else-return#rule-details) :)